### PR TITLE
Simplex cUSD integration in Valora

### PR DESCRIPTION
**Pull request by annakaz**
### Description

Add Simplex to allow users to buy cUSD. 

Simplex does not allow for an address to be passed as a parameter, so there is a banner with a user's address to copy within the Simplex webview

Update Cash In flow to go directly to the provider selection, as there is only one cUSD provider (Simplex) and one CELO provider (Moonpay) for now, so it is not necessary for users to enter an amount so we can display fees to them.

This UI will soon be replaced with the new Cash In / Out flow but wanted to land a minimal Simplex integration so we could allow our users to buy cUSD and to start testing Simplex as soon as possible. 

Screen Shot 2020-11-02 at 9 32 00 PM
Screen Shot 2020-11-02 at 9 32 09 PM
Screen Shot 2020-11-02 at 9 32 17 PM

### Tested

The integration team from Simplex tested a mainnet APK with the production integration to confirm that the purchase of cUSD works as expected
On an emulator to ensure flow displays as expected

### Related issues

- Part of #5424 

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx [hunk 1]
```diff
@@ -136,9 +133,15 @@ function FiatExchangeOptions({ route, navigation }: Props) {
       },
     ],
     addFunds: [
+      {
+        image: <Image source={simplexLogo} style={styles.simplexLogo} resizeMode={'contain'} />,
+        screen: Screens.Simplex,
+        supportedCurrenciesNote: 'cUSD only',
```
Comment (jeanregisser) [inline packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx]: This needs to be translated.

File: packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx [hunk 2]
```diff
@@ -136,9 +133,15 @@ function FiatExchangeOptions({ route, navigation }: Props) {
...
+      },
       {
         image: <Image source={moonpayLogo} style={styles.moonpayLogo} resizeMode={'contain'} />,
         screen: Screens.MoonPay,
+        supportedCurrenciesNote: 'CELO only',
```
Comment (jeanregisser) [inline packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx]: This needs to be translated.

File: packages/mobile/src/fiatExchanges/Simplex.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+function Simplex() {
+  const account = useSelector(currentAccountSelector)
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.addressBanner}>
+        <Text
+          style={[styles.tapToCopy, fontStyles.regular]}
+        >{`Tap to copy your Valora CUSD address`}</Text>
```
Comment (jeanregisser) [inline packages/mobile/src/fiatExchanges/Simplex.tsx]: This needs to be translated.

File: packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx [hunk 4]
```diff
@@ -184,6 +194,24 @@ const styles = StyleSheet.create({
     height: 30,
     width: 104,
   },
+  simplexLogo: {
+    height: 59,
+    width: 111,
+  },
```
Comment (jeanregisser) [inline packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx:200]: nit: No need to address this now, but would be nice to use a standardized height for providers logo and adjust the asset instead. The simplex logo has some padding on the left which causes it to not be aligned with the MoonPay logo on the left.
Also we should provide the @1.5x, @2x, @3x, @4x sizes so it's crisp on all devices.

File: packages/mobile/src/fiatExchanges/Simplex.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import colors from '@celo/react-components/styles/colors'
+import fontStyles from '@celo/react-components/styles/fonts'
+import * as React from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import { WebView } from 'react-native-webview'
+import { useSelector } from 'react-redux'
+import AccountNumber from 'src/components/AccountNumber'
+import { SIMPLEX_URI } from 'src/config'
```
Comment (jeanregisser) [inline packages/mobile/src/fiatExchanges/Simplex.tsx:9]: AFAIK, this new config wasn't committed, causing the build to fail.

**Review by jeanregisser — APPROVED** [review]
Great! :+1:
I went ahead and translated the new strings.

File: packages/mobile/src/fiatExchanges/Simplex.tsx (hunk 5 above)
Comment (annakaz) [reply packages/mobile/src/fiatExchanges/Simplex.tsx:9]: Updated, thanks for catching this

File: packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx (hunk 4 above)
Comment (annakaz) [reply packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx:200]: Sounds good, will merge for now as it as we'll be updating the UI with the new flow

File: packages/mobile/locales/en-US/fiatExchangeFlow.json [hunk 6]
```diff
@@ -6,6 +6,9 @@
   "cashOutComingSoon": "Cash Out (Coming Soon)",
   "celoDollarsAt": "Celo Dollars @ <1></1>",
   "amountCelo": "Amount (CELO)",
+  "onlyCeloDollars": "cUSD only",
+  "onlyCelo": "CELO only",
+  "tapToCopyCeloDollarsAddress": "Tap to copy your {{appName}} CUSD address",
```
Comment (gnardini) [inline packages/mobile/locales/en-US/fiatExchangeFlow.json:11]: Is the cap on `CUSD` (as opposed to `cUSD`) intended?

**Review by gnardini — APPROVED** [review]
Awesome to have another provider 😄

**Comment by Lss-Ankit** [comment]
Hi @annakaz , @jeanregisser  Started verifying task using v1.4.0 (28) and Android play store (internal build) v1.4.0 (1004294317) and found below new issue: 
- https://github.com/celo-org/celo-monorepo/issues/5743. 

Also, in Android [Randomly] Something went wrong error message screen is shown after tapping on Simplex option from Add and Withdraw option. 
- Not getting verification code from mail id while adding cUSD from Simplex option. 
Note: 
- Checked spam mail as well but we are not getting mails there as well. 
Please let us know if we need to file a bug as we are not getting verification code mail 
Hence we are blocked to performed testing on this task can you please do the needful.

**Pull request merged**